data-prepper-plugins/elasticsearch/src/test/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfigurationTests.java

```diff
@@ -118,6 +118,26 @@ public void testCreateClientWithAWSSigV4DefaultRegion() throws IOException {
         assertTrue(connectionConfiguration.isAwsSigv4());;
     }
 
+    @Test
+    public void testCreateClientWithAWSSigV4AndInsecure() throws IOException {
+        final PluginSetting pluginSetting = generatePluginSetting(
+                TEST_HOSTS, null, null, null, null, true, null, null, true);
+        final ConnectionConfiguration connectionConfiguration =
+                ConnectionConfiguration.readConnectionConfiguration(pluginSetting);
+        assertEquals("us-east-1", connectionConfiguration.getAwsRegion());
+        assertTrue(connectionConfiguration.isAwsSigv4());;
```

Comment:
Are there any other assertions we can make on these 2 scenarios? Currently they make the same assertions even though the code paths are different. If not then integ tests may be the way to go.

Reply:
`insecure` flag is not exposed. integ test might be the only way to manifest.